plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergOrcFileWriter.java

```diff
@@ -261,10 +276,20 @@ private static int getIcebergId(OrcType orcColumn)
         private ByteBuffer min;
         private ByteBuffer max;
 
-        private IcebergMinMax(org.apache.iceberg.types.Type type, Object min, Object max)
+        private IcebergMinMax(org.apache.iceberg.types.Type type, Object min, Object max, MetricsModes.MetricsMode metricsMode)
         {
-            this.min = Conversions.toByteBuffer(type, min);
-            this.max = Conversions.toByteBuffer(type, max);
+            // Out of the two types which could be truncated, string or binary, ORC only supports string bounds.
+            // Therefore, truncation will be applied if needed only on string type.
+            if (metricsMode instanceof MetricsModes.Truncate && type.typeId() == org.apache.iceberg.types.Type.TypeID.STRING) {
```

Comment:
binary (varbinary)  and fixed too 

https://github.com/apache/iceberg/blob/2e4684740b56f6d80328f50b7e7fef0863c262e7/parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java#L269

(we don't support creating tables with `fixed`, but i think we do support writes to such tables)

Reply:
@findepi I was referring to the `OrcMetrics` implementation here: https://github.com/apache/iceberg/blob/ddc5aff9167c7e367e2b4313b91e55451f09ef16/orc/src/main/java/org/apache/iceberg/orc/OrcMetrics.java#L281 
Seems like it only added truncation for string bound in the orc format.

Reply:
because they don't produce any stats for binary for ORC, see https://github.com/trinodb/trino/blob/88ef2caf7b5f374c7558e2f06df754c2d6b1acac/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergOrcConnectorTest.java#L31 
right?

Reply:
still, not having an `if` for varbinary doesn't seem future proof, does it?

Reply:
@findepi yes I agree. I'll add varbinary here.